src/components/Header/Header.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import bem from 'bem';
+import styles from './Header.scss';
+
+const { block, elem } = bem({
+    name: 'Header',
+    classnames: styles
+});
+
+const Header = props => {
+    const { logoSrc, logoLink, logoName, children, ...rest } = props;
+
+    return (
+        <div {...rest} {...block(props)}>
+            <div {...elem('wrapper', props)}>
+                <a href={logoLink}>
+                    <img src={logoSrc} alt={logoName} {...elem('logo', props)} />
+                </a>
+                <div {...elem('menu', props)}>{children}</div>
+            </div>
+        </div>
+    );
+};
+
+Header.displayName = 'Header';
+
+Header.propTypes = {
+    /** path to the logo source file */
+    logoSrc: PropTypes.string.isRequired,
+    /** path to where the logo should link to if clicked */
+    logoLink: PropTypes.string,
```

Comment:
What if you need to use a callback function or using React router?

Reply:
I was hoping you won't ask. If we want to handle those cases, we need to define them in more detail. I think the most common case for the logo is to just link to Home ('/'). Which should work even without react-router link. I suggest that we handle other cases when we actually come across them.

Reply:
As long as you say it can work with React router without having to reload a whole page it's fine with me for now